MobileSDK/sdk/src/main/java/org/lantern/mobilesdk/Lantern.java

```diff
@@ -44,44 +35,80 @@
      * @return the {@link go.lantern.Lantern.StartResult} with port information about the started
      * lantern
      */
-    public synchronized static go.lantern.Lantern.StartResult enable(Context context, int timeoutMillis, String analyticsTrackingId)
+    public static StartResult enable(Context context, int timeoutMillis, String analyticsTrackingId)
+            throws LanternNotRunningException {
+        return doEnable(context, timeoutMillis, analyticsTrackingId, "org.lantern.mobilesdk.embedded.EmbeddedLantern");
+    }
+
+    /**
+     * Like {@link #enable(Context, int, String)} but runs the proxy in a Service.
+     *
+     * @param context
+     * @param timeoutMillis
+     * @param analyticsTrackingId
+     * @return
+     * @throws LanternNotRunningException
+     */
+    public static StartResult enableAsService(Context context, int timeoutMillis, String analyticsTrackingId)
+            throws LanternNotRunningException {
+        return doEnable(context, timeoutMillis, analyticsTrackingId, "org.lantern.mobilesdk.LanternServiceManager");
+    }
+
+    private synchronized static StartResult doEnable(Context context, int timeoutMillis, String analyticsTrackingId, String implClassName)
             throws LanternNotRunningException {
-        String configDir = new File(context.getFilesDir().getAbsolutePath(), ".lantern").getAbsolutePath();
-        go.lantern.Lantern.StartResult result = enable(configDir, timeoutMillis);
+        Lantern lantern = instanceOf(implClassName);
+        StartResult result = lantern.start(context, timeoutMillis);
+        proxyOn(result.getHTTPAddr());
         if (analyticsTrackingId != null && !enabled) {
             trackStartSession(context, analyticsTrackingId);
         }
+        enabled = true;
         return result;
     }
 
-    private static go.lantern.Lantern.StartResult enable(String configDir, int timeoutMillis)
-            throws LanternNotRunningException {
+    /**
+     * Note - we use dynamic class loading to avoid loading unused classes into the caller's
+     * classloader (i.e. to avoid loading native dependencies when running as service). This is
+     * important because in some situations, it appears that including the Lantern native library
+     * inside the same process as an application can cause instability on some phones (e.g. Samsung
+     * Galaxy S4).
```

Comment:
This comment is important - it turns out that simply loading the native library into the Manoto app introduced instability.  I structured Lantern so that simply importing it does not cause the native library to get loaded.  That now happens only when the proxy is started, which when running as a service happens in a separate process.